spec.html

```diff
@@ -35348,9 +35758,57 @@ <h1>
+            <p>In case-insignificant matches when _EitherUnicode_ is *true*, all characters are implicitly case-folded using the simple mapping provided by the Unicode standard immediately before they are compared. The simple mapping always maps to a single code point, so it does not map, for example, `&szlig;` (U+00DF) to `SS`. It may however map a code point outside the Basic Latin range to a character within, for example, `&#x17f;` (U+017F) to `s`. Such characters are not mapped if _EitherUnicode_ is *false*. This prevents Unicode code points such as U+017F and U+212A from matching regular expressions such as `/[a-z]/i`, but they will match `/[a-z]/ui`.</p>
           </emu-note>
         </emu-clause>
+        <emu-clause id="sec-runtime-semantics-maybesimplecasefolding-a" type="abstract operation">
+          <h1>
+            MaybeSimpleCaseFolding (
+              _A_: a CharSet,
+            )
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd>It uses the Unicode <a href="https://www.unicode.org/reports/tr44/#Simple_Case_Folding">Simple Case Folding</a> mapping, which maps one code point to one code point, and which is abbreviated <a href="https://www.unicode.org/Public/UCD/latest/ucd/PropertyAliases.txt">scf</a>.</dd>
+          </dl>
+          <emu-alg>
+            1. If _UnicodeSets_ is *false* or _IgnoreCase_ is *false*, return _A_.
+            1. Let _S_ be a copy of _A_.
+            1. For each element _cs_ (a character or sequence of characters) in _A_, do
+              1. Create an empty sequence of characters _t_.
+              1. For each single code point _c_ in _cs_, do
+                1. Append scf(_c_) to _t_.
+              1. If _t_ is different from _cs_, remove _cs_ from _S_ and add _t_ to _S_.
+            1. Return _S_.
+          </emu-alg>
+        </emu-clause>
+        <emu-clause id="sec-runtime-semantics-getallcharacters" type="abstract operation">
+          <h1>
+            GetAllCharacters (
+            )
+          </h1>
+          <dl class="header">
+          </dl>
+          <emu-alg>
+            1. Let _A_ be a new, empty CharSet (“A” for “all characters”).
+            1. If _UnicodeSets_ is *true* and _IgnoreCase_ is *true*, add all Unicode code points _c_ to _A_ that do not have a <a href="https://www.unicode.org/reports/tr44/#Simple_Case_Folding">Simple Case Folding</a> mapping (that is, scf(_c_)=_c_).
+            1. Otherwise, add all characters to _A_.
+            1. Return _A_.
+          </emu-alg>
+        </emu-clause>
+        <emu-clause id="sec-runtime-semantics-codepointcomplement-s" type="abstract operation">
+          <h1>
+            CodePointComplement (
```

Comment:
`CodePointComplement` returns the code _unit_ complement in non-`u` regexes, so this is a bad name. Just `Complement` seems fine.

Reply:
Seems... broad but I am ok with renaming. Maybe `CharacterComplement` like other "CharacterSomethings" that are flag-dependent?

Reply:
`CharacterComplement` works for me.

Reply:
done: 5c731e1